source/features/single-diff-column-selection.tsx

```diff
@@ -0,0 +1,25 @@
+import './single-diff-column-selection.css';
+
+import select from 'select-dom';
+import delegate from 'delegate-it';
+
+import features from '.';
+
+function disableDiffSelection(event: delegate.Event<MouseEvent, HTMLElement>): void {
+	if (event.button === 0) {
+		document.getSelection()!.empty();
+		select('[data-rgh-select]')?.removeAttribute('data-rgh-select');
+		event.delegateTarget.closest('tbody')!.dataset.rghSelect = event.delegateTarget.closest('td:last-child') ? 'right' : 'left';
+	}
```

Comment:
It looks like only this is required:
```suggestion
	event.delegateTarget.closest('tbody')!.dataset.rghSelect = event.delegateTarget.closest('td:last-child') ? 'right' : 'left';
```

Reply:
> 1. The selection is already reset on mousedown by the browser

`document.getSelection()!.empty();` is here to prevent this annoying behavior in Chromium:

![demo](https://user-images.githubusercontent.com/46634000/98528355-79390080-227c-11eb-9083-d046a7a61f13.gif)

> 2. The column selection attribute doesn't really affect the user until the next selection, in which this listener will just override it

True, it was mostly to avoid littering the DOM and making more and more diffs un-selectable via keyboard shortcuts.

> 3. The button condition… what is it for?

To avoid resetting the selection when opening the context menu with a right-click.

Reply:
1. Indeed, however that behavior is desired when dragging the (visibly-)selected text, so we shouldn’t break it.

Reply:
I think we should listen to the more specific `selectionstart` event instead of `mousedown`. It might also fix some of the issues (e.g. no need for the condition)

Reply:
> I think we should listen to the more specific `selectstart` event instead of `mousedown`. It might also fix some of the issues (e.g. no need for the condition)

We probably still need to listen to `mousedown` since a disabled element will not throw any `selectstart` events when the user attempts to select it.